### ISSUE TITLE
More intelligent password strength

### ISSUE THREAD

**Issue by matrixbot** [element-hq/matrix-authentication-service#172]
*This issue was originally created by [**@hughns**](https://github.com/hughns) at <https://github.com/matrix-org/matrix-authentication-service/issues/172>.*

There is a basic password policy in place, which can set a minimum password length, require uppercase, lowercase and/or numeric symbols.

This is very basic and lacks direct feedback to the user.
We should instead use a real password strength estimation like zxcvbn and enforce it.
This would also need live feedback to the user whether their password is strong enough or not.

Open questions:

 - Is the "interactive feedback as you type" important? If so, this would need design input
 - Should this be only a suggestion or enforced? Should we have a default?
 - If the minimum strength is configurable, it would be of an opaque number: zxcvbn gives a score out of 4
 - Should we keep in parallel the current policy knobs?

Relevant design screens:

 - [Password entry on account creation](https://www.figma.com/file/HUysJAhv6cK6p1Pc81Fxaa/Matrix-Authentication-Service-(MAS)?type=design&node-id=4770-16667&mode=design&t=fPuEstdLnvQOOyBL-4)
 - [New password when recovering the account](https://www.figma.com/file/HUysJAhv6cK6p1Pc81Fxaa/Matrix-Authentication-Service-(MAS)?type=design&node-id=3784-3167&mode=design&t=fPuEstdLnvQOOyBL-4)

**Comment by sandhose** [comment]
Closing in favour of #4490
